pkg/extensions/sync/on_demand.go

```diff
@@ -81,281 +67,103 @@ func OneImage(ctx context.Context, cfg syncconf.Config, repoDB repodb.RepoDB,
 		return err
 	}
 
-	defer demandedImgs.delete(demandedImage)
-	defer close(imageChannel)
+	syncResult := make(chan error)
+	onDemand.requestStore.Store(req, syncResult)
+
+	defer onDemand.requestStore.Delete(req)
+	defer close(syncResult)
 
-	go syncOneImage(ctx, imageChannel, cfg, repoDB, storeController, repo, reference, artifactType, log)
+	go onDemand.syncImage(repo, reference, syncResult)
```

Comment:
Why do we need this goroutine?

Reply:
We only start this go routine on the first call to the outer function, the rest of simultaneous calls are waiting for the first one to exit